src/gen/conditions.rs

```diff
@@ -2061,7 +2074,7 @@ fn test_create_coin_negative_amount() {
         cond_test("((({h1} ({h2} (123 (((51 ({h2} (-1 )))))")
             .unwrap_err()
             .1,
-        ErrorCode::InvalidCoinAmount
+        ErrorCode::NegativeAmount
```

Comment:
+1 I would have thought this was already the error we got. This is a good improvement